pages/agent/v3/signed-pipelines.md

```diff
@@ -0,0 +1,122 @@
+- Any plugins used by the step, and their configurations
+- The matrix configuration [^1]
+- The step's key, if present
+
+[^1]: The matrix configuration is signed as a whole, rather than each individual matrix job. This means that the signature will be the same for each job in the matrix. When signatures are verified for matrix jobs, we double-check that the job we've received is a valid construction of the matrix, and that the signature matches the matrix configuration.
+
+Additionally, we also sign the following aspects of the build for each job:
+- The pipeline's UUID [WIP]
+- The organization UUID [WIP]
+- The build number [WIP]
+- The repo the build is running for [WIP]
+- The commit the build is running for [WIP]
+
+
+## Enabling signed pipelines on your agents
+
+### 1. Generating a keypair
+Behind the scenes, Signed Pipelines uses [JSON Web Signing (JWS)](https://datatracker.ietf.org/doc/html/rfc7797) to generate its signatures. This means that you'll need to generate a [JSON Web Key Set (JWKS)](https://datatracker.ietf.org/doc/html/rfc7517) to sign and verify your pipelines with, then configure your agents to use those keys.
+
+Luckily, the Buildkite Agent has you covered! There's a JWKS generation tool built into the agent, which you can use to generate a key pair for you. To run it, you'll need to [install the agent on your machine](/docs/agent/v3/installation), and then run:
+```bash
+buildkite-agent tool keygen --alg <algorithm> --key-id <key-id>
+```
+
+replacing `<algorithm>` and `<key-id>` with the algorithm you want to use, and the key ID you want to use. For example, to generate an EdDSA key pair with a key ID of `my-key-id`, you'd run:
+```bash
+buildkite-agent tool keygen --alg EdDSA --key-id my-key-id
+```
+
+The agent will then generate two JSON Web Key Sets for you, one public and one private, and output them to files in the current directory. You can then use these keys to sign and verify your pipelines.
+
+If no key id is provided, the agent will generate a random one for you.
+
+Note that the value of `--alg` must be a valid [JSON Web Signing Algorithm](https://datatracker.ietf.org/doc/html/rfc7518#section-3). The agent does not support the `none` algorithm, or any signature algorithms in the `RS` series (any signing algorithms that use RSASSA-PKCS1 v1.5 signatures, `RS256` for example).
+
+<details>
+  <summary>Why doesn't the agent support RSASSA-PKCS1 v1.5 signatures?</summary>
+  In short, it's because RSASSA-PKCS1 v1.5 signatures are generally regarded to be less secure than the newer RSA-PSS signatures. While RSASSA-PKCS1 v1.5 signatures are still largely known to be relatively secure, we want to encourage our users to use the most secure algorithms possible, so when using RSA keys, we only support RSA-PSS signatures. We also recommend looking into ECDSA and EdDSA signatures, which are generally regarded to be more secure than RSA signatures.
```

Comment:
```suggestion
  In short, it's because RSASSA-PKCS1 v1.5 signatures are generally regarded to be less secure than the newer RSA-PSS signatures. While RSASSA-PKCS1 v1.5 signatures are still largely known to be relatively secure, we want to encourage our users to use the most secure algorithms possible, so when using RSA keys, we only support RSA-PSS signatures. We also recommend looking into ECDSA and EdDSA signatures.
```
I'd leave this out because it may not be accurate under quantum attacks. It looks like because ECC uses keys that are an order of magnitude shorter, they may be broken by smaller quantum computers than that required to break RSA that would be weaker classically: see https://security.stackexchange.com/questions/33069/why-is-ecc-more-vulnerable-than-rsa-in-a-post-quantum-world

Of course, no such quantum computer is known to exist.

Reply:
We should get Ashwin's take on this all too

Reply:
yeah, stuff like this was why i hedged hard af whenever i talked about crypto - there's a lot of "generally regarded to be" and "thought to be" and "can be".

regardless, i don't mind this page to become a lie when quantum computing becomes available